src/main/java/org/troyargonauts/robot/subsystems/DriveTrain.java

```diff
@@ -1,249 +1,198 @@
+        leftSide = new MotorControllerGroup<>(frontLeft, List.of(middleLeft, backLeft), false, false);
+
+        dualSpeedTransmission = new DualSpeedTransmission(this);
 
         pigeon = new Pigeon2(Constants.DriveTrain.PIGEON);
 
-        drivePID = new PIDController(Constants.DriveTrain.kDriveP, Constants.DriveTrain.kDriveI, Constants.DriveTrain.kDriveD);
-        turnPID = new PIDController(Constants.DriveTrain.kTurnP, Constants.DriveTrain.kTurnI, Constants.DriveTrain.kTurnD);
-        autoBalancePID = new PIDController(Constants.DriveTrain.kBalanceP, Constants.DriveTrain.kBalanceI, Constants.DriveTrain.kBalanceP);
+        configMotors(rightSide);
+        configMotors(leftSide);
+
+        drivePID = new PIDController(Constants.DriveTrain.DRIVE_P, Constants.DriveTrain.DRIVE_I, Constants.DriveTrain.DRIVE_D);
+        turnPID = new PIDController(Constants.DriveTrain.TURN_P, Constants.DriveTrain.TURN_I, Constants.DriveTrain.TURN_D);
 
-        drivePID.setTolerance(Constants.DriveTrain.kDriveTolerance);
-        turnPID.setTolerance(Constants.DriveTrain.kTurnToleranceDeg);
-        autoBalancePID.setTolerance(Constants.DriveTrain.kBalanceToleranceDeg);
+        drivePID.setTolerance(Constants.DriveTrain.DRIVE_TOLERANCE, Constants.DriveTrain.VELOCITY_TOLERANCE);
+        turnPID.setTolerance(Constants.DriveTrain.TURN_TOLERANCE_DEG, Constants.DriveTrain.VELOCITY_TOLERANCE);
 
         turnPID.enableContinuousInput(-180, 180);
 
-        resetEncoders();
+        rightSide.forEach(talonFX -> talonFX.setGearingParameters(Constants.DriveTrain.gearingLowGear));
+        leftSide.forEach(talonFX -> talonFX.setGearingParameters(Constants.DriveTrain.gearingLowGear));
     }
 
     @Override
     public void periodic() {
-        frontRightEncoderValue = frontRight.getEncoder().getPosition();
-        middleRightEncoderValue = middleRight.getEncoder().getPosition();
-        backRightEncoderValue = backRight.getEncoder().getPosition();
-        frontLeftEncoderValue = frontLeft.getEncoder().getPosition();
-        middleLeftEncoderValue = middleLeft.getEncoder().getPosition();
-        backLeftEncoderValue = backLeft.getEncoder().getPosition();
-
-        SmartDashboard.putNumber("frontRightEncoderValue", frontRightEncoderValue);
-        SmartDashboard.putNumber("middleRightEncoderValue", middleRightEncoderValue);
-        SmartDashboard.putNumber("backRightEncoderValue", backRightEncoderValue);
-        SmartDashboard.putNumber("frontLeftEncoderValue", frontLeftEncoderValue);
-        SmartDashboard.putNumber("middleLeftEncoderValue", middleLeftEncoderValue);
-        SmartDashboard.putNumber("backLeftEncoderValue", backLeftEncoderValue);
-
-        gyroValue = pigeon.getYaw();
-    }
+        rightEncoderValue = rightSide.getMaster().getInternalController().getSensorCollection().getIntegratedSensorPosition();
+        leftEncoderValue = leftSide.getMaster().getInternalController().getSensorCollection().getIntegratedSensorPosition();
```

Comment:
Can we make the rightEncoderValue an average of all 3 motors from that side.

Reply:
Already had a conversation about this, refer to above.